lib/board.rb

```diff
@@ -31,31 +31,47 @@ def valid_coordinate?(coordinate)
 
   def valid_placement?(ship, coordinates)
 
-    # turns given coordinates into array of integers
-    columns = coordinates.map { |c| c[1].to_i }
-    # turns given coordinates into array of letter.ordinal_values
-    rows = coordinates.map { |c| c[0].ord }
-    # turns coordinates into array of letter strings
-    coordinate_letters = coordinates.map { |c| c[0] }
-
-    # our checks!
-    horizontal_check = (1..4).each_cons(ship.length).any?(columns)
-    # vertical check
-    letters_ord = ("A".."D").to_a.map { |a| a.ord }
-    vertical_check = letters_ord.each_cons(ship.length).any?(rows)
-
-    if ship.length == coordinates.length
-      if horizontal_check && coordinate_letters.all?(coordinate_letters[0])
-        true
-      elsif vertical_check && !coordinate_letters.all?(coordinate_letters[0]) && columns.all?(columns[0])
-        true
-      else
-        false
-      end
+    # valid number of coordinates
+    valid_length = ship.length == coordinates.length
+    # coordinate columns as integer
+    columns = coordinates.map do |coordinate|
+      coordinate[1].to_i end
+    # coordinate rows as letter string
+    rows = coordinates.map do |coordinate|
+      coordinate[0] end
+
+    # checking consecutiveness
+    horizontally_cons = (1..4).each_cons(ship.length).any?(columns)
+
+    vertically_cons = ("A".."D").each_cons(ship.length).any?(rows)
+
+    same_row = rows.all?(rows[0])
+    same_column = columns.all?(columns[0])
+
+    # valid placement conditional tree
+    # going to need to check whether cells[].empty? first
+    if valid_length && horizontally_cons && same_row
+      true
+    elsif valid_length && vertically_cons && !same_row && same_column
+      true
     else
       false
     end
+  end
 
+  def place(ship, coordinates)
```

Comment:
@amymoller I'm having trouble with this method; it's not complete, bc the tests don't pass... but when I go into pry and check whether the ship was placed on the cell object, IT HAS. So idk why it doesn't "happen" when I run the test (lines 97-99); it still thinks cell.ship == nil

Reply:
@amymoller fixed it!